File: frontend/src/services/ratingService.ts
import api from './api';
import { Rating } from '../types';

export interface VendorRatingsResponse {
  averageScore: number;
  totalRatings: number;
  ratings: Rating[];
}

export async function submitRating(payload: {
  orderId: string;
  score: number;
  feedback?: string;
}) {
  const response = await api.post('/ratings', payload);
  return response.data;
}

export async function getVendorRatings(
  vendorId: string,
): Promise<VendorRatingsResponse> {
  const response = await api.get<VendorRatingsResponse>(`/vendors/${vendorId}/ratings`);
  return response.data;
}

File: flutter_geobite/lib/services/rating_service.dart
import '../models/rating.dart';
import '../core/api_client.dart';

class RatingService {
  Future<void> submitRating(String orderId, int score, String? feedback) async {
    try {
      await apiClient.dio.post(
        '/ratings',
        data: {
          'orderId': orderId,
          'score': score,
          'feedback': feedback,
        },
      );
    } catch (e) {
      throw Exception('Failed to submit rating: $e');
    }
  }

  Future<VendorRatingSummary> getVendorRatings(String vendorId) async {
    try {
      final response = await apiClient.dio.get('/vendors/$vendorId/ratings');
      return VendorRatingSummary.fromJson(response.data);
    } catch (e) {
      throw Exception('Failed to fetch ratings: $e');
    }
  }
}

final ratingService = RatingService();
